### ISSUE TITLE
Is DeepFace differentiable, i.e., able to back-propagate gradients to update input?

### ISSUE THREAD

**Issue by BannyStone** [serengil/deepface#512]
I want to use deepface to analyze our synthesized images. The loss is computed based on deepface's output. And we want to bp the loss gradient through deepface to update the synthesizer. Is it possible?

**Comment by serengil** [comment]
yes, deepface models are actually keras models except dlib and sface. so, you are able to run model.train for those keras models.

**Comment by Parul-Gupta** [comment]
@serengil I have been facing an issue with this: my image synthesizer outputs pytorch tensors which are then fed into the deepface represent model. However, DeepFace.represent() accepts only numpy arrays as input. Now, converting a pytorch tensor to numpy array can only occur when its requires_grad is False, thus breaking the gradient flow for back propagation. Any suggestions?

**Comment by serengil** [comment]
@Parul-Gupta no experience on pytorch, sorry.
